### PR TITLE
Try to generalize KBStorageManager

### PR DESCRIPTION
**Pull request by ChandlerYan**
Use an ExpirableEntityType instead of only KBAsset

### CODE REVIEW

File: ExpirableEntityType.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//  ExpirableEntityType.swift
+//  KikBank
+//
+//  Created by Yucheng Yan on 2018/3/20.
+//
+
+import Foundation
+
+@objc public protocol ExpirableEntityType {
+    var key: String { get set }
```
Comment (JamesRagnar) [inline ExpirableEntityType.swift]: Is there a reason the key should change?

File: ExpirableEntityType.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
```
Comment (JamesRagnar) [inline ExpirableEntityType.swift]: File is in the project root, move to `KikBank/Classes`

File: KikBank/Classes/KBStorageManager.swift [hunk 3]
```diff
@@ -13,102 +13,108 @@ import Foundation
...
 
-    private struct Constants {
-        static let cachePathExtension = "KikBankStorage"
+    var cachePathExtension: String {
+        get {
+            return self.cachePathExtension
+        }
+        
+        set {
+            self.cachePathExtension = "/" + newValue
```
Comment (JamesRagnar) [inline KikBank/Classes/KBStorageManager.swift]: What is the purpose of this?

File: KikBank/Classes/KBAsset.swift [hunk 4]
```diff
@@ -8,37 +8,36 @@
 import Foundation
 
 /// Cached and saved object which tracks data validity
-class KBAsset: NSObject {
-
+@objc public class KBAsset: NSObject, ExpirableEntityType {
```
Comment (JamesRagnar) [inline KikBank/Classes/KBAsset.swift]: Like the idea of having the `Expirable` type, but that might duplicate some properties down the line. Keep the `key` and `isValid` properties as part of the more generic `KBAssetType`, and let `ExpireableEntityType` worry about handling the date

File: ExpirableEntityType.swift [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+//  ExpirableEntityType.swift
+//  KikBank
+//
+//  Created by Yucheng Yan on 2018/3/20.
+//
+
+import Foundation
+
+@objc public protocol ExpirableEntityType {
```
Comment (JamesRagnar) [inline ExpirableEntityType.swift]: Follow `KB` naming scheme for now, we will review that in a later build

File: KikBank/Classes/KBAsset.swift (hunk 4 above)
Comment (ChandlerYan) [reply KikBank/Classes/KBAsset.swift]: @JamesRagnar I think isValid is related to expiryDate, and we should keep this property in ExpirableEntityType.

**Pull request merged**